src/vendor/pg_ruleutils_17.c

```diff
@@ -12493,26 +12493,33 @@ static void
 get_tablesample_def(TableSampleClause *tablesample, deparse_context *context)
 {
 	StringInfo	buf = context->buf;
-	Oid			argtypes[1];
+	Oid			argtypes[1] = { INTERNALOID };
+	char 		*tsm_name;
 	int			nargs;
 	ListCell   *l;
 
 	/*
 	 * We should qualify the handler's function name if it wouldn't be
 	 * resolved by lookup in the current search path.
 	 */
-	argtypes[0] = INTERNALOID;
-	appendStringInfo(buf, " TABLESAMPLE %s (",
-					 generate_function_name(tablesample->tsmhandler, 1,
-											NIL, argtypes,
-											false, NULL, EXPR_KIND_NONE));
+	tsm_name = generate_function_name(tablesample->tsmhandler, 1, NIL, argtypes,
+	false, NULL, EXPR_KIND_NONE);
+	appendStringInfo(buf, " TABLESAMPLE %s (", tsm_name);
 
-	nargs = 0;
-	foreach(l, tablesample->args)
+	if (pg_strcasecmp(tsm_name, "system") == 0 && list_length(tablesample->args) == 1)
 	{
-		if (nargs++ > 0)
-			appendStringInfoString(buf, ", ");
-		get_rule_expr((Node *) lfirst(l), context, false);
+		get_rule_expr((Node *) linitial(tablesample->args), context, false);
```

Comment:
These changes also need to be made in the other `pg_ruleutils_1x.c` files. And probably we want to share some of this code to `pgduckdb_ruleutils.cpp` so the code can be shared between these files.